pynamodb/attributes.py

```diff
@@ -406,6 +406,7 @@ def __init__(self, hash_key=False, range_key=False, null=None, default=None, att
                                            null=null,
                                            default=default,
                                            attr_name=attr_name)
+        self._is_raw_map = bool(type(self) == MapAttribute)
```

Comment:
Yep, I understand that this seems to be a bit of a gray area there. My feeling is that we're getting back references to a class which is _not_ one of the constants, so they should be compared with `==` incase there is some situation where they're stored separately in memory (I don't know of a case where this happens off hand, but it seems like it plausibly could happen)